src/autorenew-invite/autorenew-invite.controller.js

```diff
@@ -0,0 +1,83 @@
+import _ from 'lodash';
+import moment from 'moment';
+
+export default class {
+  /* ngInject */
+  constructor(
+    $q,
+    OvhApiMe,
+    ovhPaymentMethod,
+    ovhUserPref,
+    WUC_SUBSIDIARIES_WITH_OPTIONAL_AUTORENEW,
+  ) {
+    this.$q = $q;
+    this.OvhApiMe = OvhApiMe;
+    this.ovhPaymentMethod = ovhPaymentMethod;
+    this.ovhUserPref = ovhUserPref;
+
+    this.WUC_SUBSIDIARIES_WITH_OPTIONAL_AUTORENEW = WUC_SUBSIDIARIES_WITH_OPTIONAL_AUTORENEW;
+  }
+
+  $onInit() {
+    this.preferenceKey = `${this.productType.toUpperCase()}_AUTORENEW_STOP_BOTHER`;
+
+    return this.isAutorenewAllowed()
+      .then(() => (this.isAutorenewAllowed ? this.getDisplayConditions() : this.$q.when()));
+  }
+
+  getDisplayConditions() {
+    return this.getProductAutorenewBotherPreferences()
+      .then(() => this.hasPaymentMean());
+  }
+
+  hasPaymentMean() {
+    return this.ovhPaymentMethod.getAllPaymentMethods()
+      .then((paymentMethods) => {
+        this.hasPaymentMean = !_.isEmpty(paymentMethods);
+        return this.hasPaymentMean;
+      });
+  }
+
+  isAutorenewAllowed() {
+    return this.OvhApiMe.v6().get().$promise
+      .then(({ ovhSubsidiary }) => {
+        this.isAutorenewAllowed = _.includes(
+          this.WUC_SUBSIDIARIES_WITH_OPTIONAL_AUTORENEW,
+          ovhSubsidiary,
+        );
+      })
+      .catch(() => {
+        this.isAutorenewAllowed = false;
```

Comment:
Looks like we should display an error message in this case

Reply:
@FredericEspiau I agree that user should know an error has occured,  but I'm not sure it is relevant in that specific case (display a message to inform that we cannot know if we should display another message or not)